var/spack/repos/builtin/packages/mfem/package.py

```diff
@@ -191,23 +216,32 @@ class Mfem(Package):
     depends_on('mpfr', when='+mpfr')
     depends_on('netcdf-c@4.1.3:', when='+netcdf')
     depends_on('unwind', when='+libunwind')
-    depends_on('zlib', when='+gzstream')
+    depends_on('zlib', when='+zlib')
     depends_on('gnutls', when='+gnutls')
     depends_on('conduit@0.3.1:,master:', when='+conduit')
     depends_on('conduit+mpi', when='+conduit+mpi')
 
     # The MFEM 4.0.0 SuperLU interface fails when using hypre@2.16.0 and
     # superlu-dist@6.1.1. See https://github.com/mfem/mfem/issues/983.
-    conflicts('+hypre+superlu-dist',
-              when='mfem@4.0.0 ^hypre@2.16.0 ^superlu-dist@6.1.1')
+    # This issue was resolved in v4.1.
+    conflicts('+superlu-dist',
+              when='mfem@:4.0.99 ^hypre@2.16.0: ^superlu-dist@6:')
```

Comment:
Any particular reason for this change?

Its breaking xsdk@0.5.0 build - which uses mfem@4.0.1-xsdk. Is this change intended to apply to this verison aswell?

Reply:
I double checked: the comment is correct the issue was resolved in `mfem@4.1.0` after `mfem@4.0.1-xsdk` (see https://github.com/mfem/mfem/pull/988).

I think this was a hidden issue: `mfem@4.0.1-xsdk` can be built with this combination of `hypre` and `superlu-dist`, however, it will crash on use.

Maybe we can remove `+superlu-dist` from https://github.com/spack/spack/blob/eef75ddd45f7d5a268cbf9eafd3af5143c271391/var/spack/repos/builtin/packages/xsdk/package.py#L47

The other option is to revert the conflict to not apply to `mfem@4.0.1-xsdk` but that is actually incorrect.

By the way, the `+hypre` variant in the mfem spec can be removed since `hypre` is a dependency due to `+mpi`.

Reply:
@v-dobrev Thanks! will make a change to xsdk/package.py and remove `+superlu-dist`